Cloudtrail log request id and request name.

goughcor opened the pull request:
*Description of changes:*
Log request name and request id.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

rjlohan commented:
> I added it to injectCredentialsAndInvoke in the latest commit. What about the other methods? (Ex: injectCredentialsAndInvokeV2InputStream) Does it need to be added there too?

Yeah you'll probably want to wrap all the calls like `return requestFunction.apply(wrappedRequest);` to get full coverage from this logging.

goughcor commented:
> > I added it to injectCredentialsAndInvoke in the latest commit. What about the other methods? (Ex: injectCredentialsAndInvokeV2InputStream) Does it need to be added there too?
> 
> Yeah you'll probably want to wrap all the calls like `return requestFunction.apply(wrappedRequest);` to get full coverage from this logging.

I added logging to the other calls.  We should have full coverage now.
